tests/integration/models/device.spec.coffee

```diff
@@ -1028,6 +1028,25 @@ describe 'Device Model', ->
 					m.chai.expect(_.find(result, { name: 'A' }).value).equal('a')
 					m.chai.expect(_.find(result, { name: 'B' }).value).equal('b')
 
+		describe 'balena.models.device.getSupervisorTargetState()', ->
+
+			it 'should be rejected if the device does not exist', ->
+				promise = balena.models.device.getSupervisorTargetState('asdfghjkl')
+				m.chai.expect(promise).to.be.rejectedWith('Device not found: asdfghjkl')
+
+			it 'should reflect the device\'s name', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) ->
+					m.chai.expect(state.local.name).to.be.a('string')
+
+			it 'should reflect the device\'s apps', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) =>
+					m.chai.expect(state.local.apps).to.be.an('object')
+					m.chai.expect(state.local.apps[@application.id.toString()].name).to.equal(@application.app_name)
```

Comment:
You probably can drop the `.toString()`s in here to make it a bit shorter, since JS will do that for you.